td-agent/Rakefile

```diff
@@ -544,6 +614,70 @@ class BuildTask
     end
   end
 
+  def openssldir
+    File.join(staging_etcdir, "openssl")
+  end
+
+  def create_certs
+    keychains = [
+      "/System/Library/Keychains/SystemRootCertificates.keychain"
+    ]
+
+    cert_list = `security find-certificate -a -p #{keychains.join(" ")}`
+    certs = cert_list.scan(
+      /-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----/m
+    )
+
+    valid_certificates = certs.select do |cert|
+      IO.popen("#{staging_bindir}/openssl x509 -inform pem -checkend 0 -noout > /dev/null", "w") do |ossl_io|
+        ossl_io.write(cert)
+        ossl_io.close_write
+      end
+
+      $?.success?
+    end
+
+    mkdir_p(openssldir)
+    File.open(File.join(openssldir, "cert.pem"), 'w', 0644) do |f|
+      f.write(valid_certificates.join("\n") << "\n")
+    end
+  end
+
+  def build_openssl
+    tarball = @download_task.file_openssl_source
+    source_dir = tarball.sub(/\.tar\.gz$/, '')
+
```

Comment:
File.basename(tarball, ".tar.gz") ?

Reply:
No. The suggested path should not work.
file_openssl_source should be included DOWNLOADS_DIR.
```
@file_openssl_source = File.join(DOWNLOADS_DIR, filename)
```

But your suggestion should drop DOWNLOADS_DIR from source_dir value.

Reply:
Ah, I've overlooked it. :sob: